tools/apiview/parsers/apiview-treestyle-parser-schema/models.tsp

```diff
@@ -0,0 +1,150 @@
+model CodeFile {
+  /** The version of the parser */
+  VersionString: string;
+
+  /** Name the ocde file. Usually just the package name and version*/
+  Name: string;
+  
+  Language: string;
+  LanguageVariant: "Default" | "Spring" | "Android";
+  PackageName: string;
+  ServiceName: string;
+  PackageVersion: string;
+
+  /** Maps related packages across Languages */
+  CrossLanguagePackageId: string;
+
+  /** The API Tree(s) */
+  APIForest: APITreeNode[];
+
+  Diagnostics: CodeDiagnostic[];
+}
+
+model CodeDiagnostic {
+  DiagnosticId: string;
+  Text: string;
+  HelpLinkUri: string;
+
+  /** Maps to the APITreeNode id */
+  TargetId: string;
+  Level: CodeDiagnosticLevel;
+}
+
+model APITreeNode {
+  /** The name of the tree node which will be used as label for the API Navigation. Generally this is the name of the module (class, method). */
+  Name: string;
+
+  /** Id of the node, which should be unique at the node level. i.e. unique among its siblings. This was previously represented by the DefinitionId for the main Token of the node. */
+  Id: string;
+
+  /** The kind of node */
+  Kind: "assembly" | "namespace" | "class" | "delegate" | "enum" | "interface" | "method" | "package" | "struct" | "type";
+
+  /** Tokens which are rendered after all child nodes. Depending on the language this would include the closing curly brace and/or empty lines. */
+  BottomTokens?: StructuredToken[];
+
+  /** The nodes immediate children. For a namespace this would be classes, for a class this would be the class constructors and methods. Children are rendered after TopTokens but before BottomTokens, and are automatically indented.*/
+  Children?: APITreeNode[];
+
+  /** Properties of the APITreeNode. */
+  Properties?: APITreeNodeProperties;
+
+  /** Tags of the APITreeNode. */
+  Tags?: APITreeNodeTags[];
+}
+
+model StructuredToken {
+  /** Token Id. Previously known as DefinitionId. */
+  Id: string;
+
+  /** Represents the type of a structured token*/
+  Kind: StructuredTokenKind;
+
+  /** The token value which will be displayed. Spacing tokens (LineBreak, NonBreakingSpace, TabSpace, and ParameterSeparator) don't need to have value */
+  Value?: string;
+
+  /** Properties of the StructuredToken. */
+  Properties?: StructuredTokenProperties;
+
+  /** Tags of the StructuredToken. */
+  Tags?: StructuredTokenTags[];
+
+  /** Classes used to render the StructuredTokens. */
+  RenderClasses?: StructuredTokenRenderClasses[]
+}
+
+model APITreeNodeProperties {
```

Comment:
The contributions guidelines gives example of `SubKind`, which I don't see documented here

Reply:
See https://github.com/chidozieononiwu/azure-sdk-tools/blob/main/tools/apiview/parsers/CONTRIBUTING.md#navigation

![image](https://github.com/user-attachments/assets/de485408-1713-4c2f-8b1c-9672d034ee5b)

It's unclear reading the doc, or the TypeSpec, what is the difference between a `Kind` and a `SubKind`

Reply:
I think I will remove this as Travis pointed out between Kind and the IconName name this might not be necessary. I added o that languages can add better descriptions for the node but I can remove it.